src/client/parser.rs

```diff
@@ -561,15 +561,20 @@ impl<M, S> Protocol for Parser<M, S>
     {
         use self::ParserImpl::*;
         match self.1 {
+            // skip the event, will child state machine when connected
+            me@Connecting(..) => me.intent(self.0, scope),
+            // skip the event, will child state machine when connected
+            me@Flushing(..) => me.intent(self.0, scope),
             Idle(..) => {
                 // TODO(tailhook) propagate same idle deadline
                 maybe_new_request(transport,
                     self.0.wakeup(&Connection {
                         idle: true,
                     }, scope), scope)
-            },
-            // skip the event, will child state machine when idle
-            me@_ => me.intent(self.0, scope)
+            }
+            _ => {
+                unimplemented!();
```

Comment:
@tailhook I'm not sure how I can get a reference to a `Request` here to pass to the machine's `wakeup`?